coordinator/src/test/scala/com/socrata/datacoordinator/service/collocation/CoordinatedCollocatorTest.scala

```diff
@@ -0,0 +1,210 @@
+                                         andNoOtherCalls: Boolean,
+                                         params: (String, Set[DatasetInternalName],  Set[DatasetInternalName])*): Unit = {
+    params.foreach { case (instance, datasets, result) =>
+      (mock.collocatedDatasetsOnInstance _).expects(instance, datasets).once.returning(collocatedDatasets(result))
+    }
+
+    if (andNoOtherCalls) {
+      (mock.rollbackSecondaryMoveJob _).expects(*, *, *, *).never
+      (mock.ensureSecondaryMoveJob _).expects(*, *, *).never
+      (mock.secondariesOfDataset _).expects(*).never
+      (mock.secondaryMoveJobs _).expects(*, *).never
+    }
+  }
+
+  def expectNoMockCollocationLockCalls(mock: CollocationLock): Unit = {
+    (mock.release _).expects().never
+    (mock.acquire _).expects(*).never
+  }
+
+  def withMocks[T](defaultStoreGroups: Set[String],
+                        coordinatorExpectations: Coordinator => Unit = expectNoMockCoordinatorCalls,
+                        collocationLockExpectations: CollocationLock => Unit = expectNoMockCollocationLockCalls)
+                       (f: (Collocator, CollocationManifest) => T): T = {
+    val manifest = new CollocationManifest
+
+    val mockCoordinator: Coordinator = mock[Coordinator]
+    val mockCollocationLock: CollocationLock = mock[CollocationLock]
+
+    val collocator = new CoordinatedCollocator(
+      collocationGroup = Set(alpha, bravo, charlie),
+      defaultStoreGroups = defaultStoreGroups,
+      coordinator = mockCoordinator,
+      addCollocations = manifest.add,
+      lock = mockCollocationLock,
+      lockTimeoutMillis = FiniteDuration(10, "seconds").toMillis
+    )
+
+    coordinatorExpectations(mockCoordinator)
+    collocationLockExpectations(mockCollocationLock)
+
+    f(collocator, manifest)
+  }
+
+  val storeGroupA = "store_group_a"
+  val storeGroupB = "store_group_b"
+
+  val defaultStoreGroups = Set(storeGroupA, storeGroupB)
+
+  def internalName(instance: String, datasetId: Long): DatasetInternalName =
+    DatasetInternalName(instance, new DatasetId(datasetId))
+
+  val alpha1 = internalName(alpha, 1L)
+  val alpha2 = internalName(alpha, 2L)
+
+  val bravo1 = internalName(bravo, 1L)
+  val bravo2 = internalName(bravo, 2L)
+
+  val charlie1 = internalName(charlie, 1L)
+  val charlie2 = internalName(charlie, 2L)
+
+  val datasetsEmpty = Set.empty[DatasetInternalName]
+
+  def collocatedDatasets(datasets: Set[DatasetInternalName]) = Right(CollocatedDatasetsResult(datasets))
+
+  val collocatedDatasetsEmpty = collocatedDatasets(datasetsEmpty)
+
+  // tests for collocatedDatasets(datasets: Set[DatasetInternalName]): Either[RequestError, CollocatedDatasetsResult]
+  test("collocatedDatasets should return the empty set when given the empty set") {
+    withMocks(defaultStoreGroups, { coordinator =>
+      expectCollocatedDatasetsOnInstance(coordinator, andNoOtherCalls = true,
+        (alpha, datasetsEmpty, datasetsEmpty),
```

Comment:
It's probably worth pulling your mocked setups into some vals since you could easily reuse them when implementing the rest of the tests.